xbmc/favourites/FavouritesService.cpp

```diff
@@ -0,0 +1,228 @@
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Kodi; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "FavouritesService.h"
+#include "filesystem/File.h"
+#include "Util.h"
+#include "interfaces/AnnouncementManager.h"
+#include "profiles/ProfilesManager.h"
+#include "FileItem.h"
+#include "utils/XBMCTinyXML.h"
+#include "utils/log.h"
+#include "utils/StringUtils.h"
+#include "utils/URIUtils.h"
+#include "settings/AdvancedSettings.h"
+#include "video/VideoInfoTag.h"
+#include "music/tags/MusicInfoTag.h"
+#include "URL.h"
+
+
+static bool LoadFromFile(const std::string& strPath, CFileItemList& items)
+{
+  CXBMCTinyXML doc;
+  if (!doc.LoadFile(strPath))
+  {
+    CLog::Log(LOGERROR, "Unable to load %s (row %i column %i)", strPath.c_str(), doc.Row(), doc.Column());
+    return false;
+  }
+  TiXmlElement *root = doc.RootElement();
+  if (!root || strcmp(root->Value(), "favourites"))
+  {
+    CLog::Log(LOGERROR, "Favourites.xml doesn't contain the <favourites> root element");
+    return false;
+  }
+
+  TiXmlElement *favourite = root->FirstChildElement("favourite");
+  while (favourite)
+  {
+    // format:
+    // <favourite name="Cool Video" thumb="foo.jpg">PlayMedia(c:\videos\cool_video.avi)</favourite>
+    // <favourite name="My Album" thumb="bar.tbn">ActivateWindow(MyMusic,c:\music\my album)</favourite>
+    // <favourite name="Apple Movie Trailers" thumb="path_to_thumb.png">RunScript(special://xbmc/scripts/apple movie trailers/default.py)</favourite>
+    const char *name = favourite->Attribute("name");
+    const char *thumb = favourite->Attribute("thumb");
+    if (name && favourite->FirstChild())
+    {
+      CURL url;
+      url.SetProtocol("favourites");
+      url.SetHostName(CURL::Encode(favourite->FirstChild()->Value()));
+      const std::string favURL(url.Get());
+      if (!items.Contains(favURL))
+      {
+        const CFileItemPtr item(std::make_shared<CFileItem>(name));
+        item->SetPath(favURL);
+        if (thumb)
+          item->SetArt("thumb", thumb);
+        items.Add(item);
+      }
+    }
+    favourite = favourite->NextSiblingElement("favourite");
+  }
+  return true;
+}
+
+CFavouritesService::CFavouritesService(std::string userDataFolder) : m_userDataFolder(std::move(userDataFolder))
+{
+  CFileItemList items;
+  std::string favourites = "special://xbmc/system/favourites.xml";
+  if(XFILE::CFile::Exists(favourites))
+    LoadFromFile(favourites, m_favourites);
+  else
+    CLog::Log(LOGDEBUG, "CFavourites::Load - no system favourites found, skipping");
+
+  favourites = URIUtils::AddFileToFolder(m_userDataFolder, "favourites.xml");
+  if(XFILE::CFile::Exists(favourites))
+    LoadFromFile(favourites, m_favourites);
+  else
+    CLog::Log(LOGDEBUG, "CFavourites::Load - no userdata favourites found, skipping");
+}
+
+CFavouritesService::~CFavouritesService()
+{
+}
```

Comment:
if the dtor is empty, I would declare it as <code>default</code> in the header file.